### PR TITLE
Modified IPC code.

### PR DESCRIPTION
**Pull request by djpetti**
This includes some simple refactoring, as well as an improved version of SingleQueue that uses shared state instead of an actual Queue. This is nice because a real Queue is overkill in this case.

### CODE REVIEW

**Comment by djpetti** [comment]
Sorry about the excess of commits. I rewrote history because I accidentally configured Travis incorrectly. Anyway, the merge should still be clean, just ignore everything except the last three commits.

File: messaging.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    """
+      Lockable queue with only one element
+    """
+
+    # The size of the array that stores data.
+    _SIZE = 1024
+
+    def __init__(self, name):
+      # Where we write the element.
+      self.__state = Array('c', self._SIZE)
```
Comment (wkronmiller) [inline messaging.py:32]: Not a queue. Why was this change necessary?

File: tests/test_messaging.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    self.test_queue.set("test message")
+    self.assertEqual("test message", self.test_queue.get())
+
+    # It should have nothing in it again.
+    self.assertEqual(None, self.test_queue.get())
+
+  def test_overwrite(self):
+    """ Tests that it behaves correctly when a value is already set. """
+    self.test_queue.set("The Honorable Dr. Shirley Ann Jackson")
+    self.test_queue.set("Shurlz")
```
Comment (wkronmiller) [inline tests/test_messaging.py]: lulz

Comment (djpetti) [inline tests/test_messaging.py]: :)

File: messaging.py (hunk 1 above)
Comment (djpetti) [inline messaging.py:32]: Well, it was never really a queue TBH. It wasn't necessary, but I figured that using an actual queue to implement a much simpler data structure was overkill.
